### ISSUE TITLE
ekg2 livecheck issue

### ISSUE THREAD

**Issue by chenrui333** [Homebrew/homebrew-livecheck#408]
```
$ brew livecheck ekg2
ekg2 (guessed) : 0.3.1 ==> 2_0.3.1
```

<details>
<summary>brew livecheck ekg2 --debug</summary>

```
Trying with url https://src.fedoraproject.org/lookaside/extras/ekg2/ekg2-0.3.1.tar.gz/68fc05b432c34622df6561eaabef5a40/ekg2-0.3.1.tar.gz
Trying with url https://web.archive.org/web/20161227025528/pl.ekg2.org/ekg2-0.3.1.tar.gz
Trying with url https://github.com/ekg2/ekg2
Possible git repo detected at https://github.com/ekg2/ekg2.git
ekg2_0-1-0 => #<Version:0x00007fe3d40faec8 @version="2_0-1-0", @tokens=[#<Version::NumericToken 2>, #<Version::NumericToken 0>, #<Version::NumericToken 1>, #<Version::NumericToken 0>]>
ekg2_0-1-1 => #<Version:0x00007fe3d40fadd8 @version="2_0-1-1", @tokens=[#<Version::NumericToken 2>, #<Version::NumericToken 0>, #<Version::NumericToken 1>, #<Version::NumericToken 1>]>
ekg2_0-1_branch_root => #<Version:0x00007fe3d40fad10 @version="2_0-1_branch_root", @tokens=[#<Version::NumericToken 2>, #<Version::NumericToken 0>, #<Version::NumericToken 1>, #<Version::StringToken "branch">, #<Version::StringToken "root">]>
ekg2_0-2_branch_root => #<Version:0x00007fe3d40fac48 @version="2_0-2_branch_root", @tokens=[#<Version::NumericToken 2>, #<Version::NumericToken 0>, #<Version::NumericToken 2>, #<Version::StringToken "branch">, #<Version::StringToken "root">]>
ekg2_0.3.0 => #<Version:0x00007fe3d40fab30 @version="2_0.3.0", @tokens=[#<Version::NumericToken 2>, #<Version::NumericToken 0>, #<Version::NumericToken 3>, #<Version::NumericToken 0>]>
ekg2_0.3.1 => #<Version:0x00007fe3d40fa540 @version="2_0.3.1", @tokens=[#<Version::NumericToken 2>, #<Version::NumericToken 0>, #<Version::NumericToken 3>, #<Version::NumericToken 1>]>
ekg2 (guessed) : 0.3.1 ==> 2_0.3.1
```

</details>

**Comment by samford** [comment]
For anyone reading this, you get the output above with a livecheckable like this (which doesn't exist yet):

```ruby
class Ekg2
  livecheck :url   => "https://github.com/ekg2/ekg2.git",
            :regex => /^ekg2_(\d+(?:\.\d+)+)$/
end
```

This issue occurs when using the Git strategy and the tags start with a name ending in a digit (e.g., `ekg2_0.3.1`). I have an idea of how to fix this, so I'll work on it when I get a chance.

**Comment by chenrui333** [comment]
Sounds good!

**Comment by samford** [comment]
I created a PR (#444) that addresses the underlying issue here when getting versions from Git tags. Once #444 is merged, #443 will add a livecheckable for `ekg2` and it should work as expected.
